crates/stc_ts_file_analyzer/src/analyzer/scope/vars.rs

```diff
@@ -357,9 +363,144 @@ impl Analyzer<'_, '_> {
                                 .freezed();
 
                             // TODO(kdy1): actual_ty
+
                             self.add_vars(elem, elem_ty, None, default_elem_ty, opts)?;
                         }
                     }
+                    // Type inference for functions
+                    let default_ty = match default_ty {
+                        Some(d_ty) => {
+                            let d_ty = d_ty.fold_with(&mut Widen { tuple_to_array: false });
+
+                            match d_ty {
+                                Type::Tuple(mut ty) => {
+                                    let right_type_len = match ty.elems.len() {
+                                        0 => 1,
+                                        _ => ty.elems.len(),
+                                    };
+
+                                    for (i, left_element) in arr.elems.iter().enumerate() {
+                                        let is_not_assigned_type = i > right_type_len - 1;
```

Comment:
You should do `i + 1> right_type_len` instead

Reply:
Oh, that makes sense